src/radical/pilot/agent/scheduler/base.py

```diff
@@ -471,16 +477,21 @@ def _schedule_units(self, units):
         # advance state, publish state change, do not push unit out.
         self.advance(units, rps.AGENT_SCHEDULING, publish=True, push=False)
 
-      # # sort units by size
-      # def _sort(a,b):
-      #     da = a['description']
-      #     db = b['description']
-      #     va = da['cpu_processes'] * da['cpu_threads'] + da['gpu_processes']
-      #     vb = db['cpu_processes'] * db['cpu_threads'] + db['gpu_processes']
-      #     return cmp(vb, va)
-      # units.sort(_sort)
-
+        # make sure we have a unit size estimate
```

Comment:
We need tests for this method(s)

Reply:
I'll revert the scheduler changes - those will be properly applied when merging the `feature/sched_nolock` branch.